apps/rush-lib/src/logic/setup/TerminalInput.ts

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as readline from 'readline';
+import * as process from 'process';
+import colors from 'colors';
+
+import { KeyboardLoop } from './KeyboardLoop';
+import { AnsiEscape } from '@rushstack/node-core-library';
+
+export interface IBasePromptOptions {
+  question: string;
+}
+
+export interface IPromptYesNoOptions extends IBasePromptOptions {
+  defaultValue?: boolean | undefined;
+}
+
+export interface IPromptPasswordOptions extends IBasePromptOptions {
+  /**
+   * The string length must not be longer than 1.  An empty string means to show the input text.
+   * @defaultValue `*`
+   */
+  passwordCharacter?: string;
```

Comment:
Terminal password input usually just doesn't print anything while the user is typing.

Reply:
The other prompt libraries print stars.  If it doesn't print stars, it's a little confusing, for example in the `rush setup` workflow the input is copy+pasted from the clipboard and it really helps to have some confirmation that it got pasted correctly.  (If not, then you have to redo the interaction from step 1.)

Also, from a security perspective, we probably are not TOO concerned about a malicious person watching over your shoulder as you are typing.  But we would be concerned about a log window persisting the input.  And we don't want the `readline()` history to remember your input.

🤔 As a compromise, maybe after you press ENTER we could delete the stars so that the password length is not recorded.

While you are typing:
```
What is the password? **************_
```

After you press ENTER it shows only 3 stars:
```
What is the password? ***
```

Reply:
Fixed